File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] ar = s.toCharArray();
        int start = 0;
        int end = s.length()-1;
        char temp;

        while(start<end){
            while(start<end && !isVowel(ar[start])) start++;
            while(start<end && !isVowel(ar[end])) end--;

            if(start<end){
                temp = ar[start];
                ar[start] = ar[end];
                ar[end] = temp;
                start++;
                end--;
            }
        }
        return new String(ar);
    }

    private boolean isVowel(char c){
        return (c=='a' || c=='e' || c=='i' || c=='o' || c=='u' || c=='A' || c== 'E' || c== 'I' || c=='O' || c=='U');
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        word = list(s)
        start = 0
        end = len(s)-1
        while start<end:
            while start<end and not self.isVowel(word[start]):
                start +=1
            while start<end and not self.isVowel(word[end]):
                end -=1
            if start<end:
                word[start], word[end] = word[end], word[start]
                start += 1
                end -= 1
        return "".join(word)
    def isVowel(self,c):
        return (c=='a' or c=='e' or c=='i' or c=='o' or c=='u' or c=='A' or c== 'E' or c== 'I' or c=='O' or c=='U')